internal/provider/resources/resource_api_token_test.go

```diff
@@ -0,0 +1,568 @@
+package resources_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/astronomer/terraform-provider-astro/internal/clients"
+	"github.com/astronomer/terraform-provider-astro/internal/clients/iam"
+	astronomerprovider "github.com/astronomer/terraform-provider-astro/internal/provider"
+	"github.com/astronomer/terraform-provider-astro/internal/utils"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/assert"
+)
+
+// Acceptance tests for Organization, Workspace, and Deployment API tokens
+// Within each test we create, update, change resource type and import the resource
+
+func TestAcc_ResourceOrganizationApiToken(t *testing.T) {
+	namePrefix := utils.GenerateTestResourceName(10)
+
+	organizationId := os.Getenv("HOSTED_ORGANIZATION_ID")
+	workspaceId := os.Getenv("HOSTED_WORKSPACE_ID")
+	deploymentId := os.Getenv("HOSTED_DEPLOYMENT_ID")
+
+	apiTokenName := fmt.Sprintf("%v_org", namePrefix)
+	resourceVar := fmt.Sprintf("astro_api_token.%v", apiTokenName)
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: astronomerprovider.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { astronomerprovider.TestAccPreCheck(t) },
+		CheckDestroy: resource.ComposeTestCheckFunc(
+			// Check that the organization api token has been removed
+			testAccCheckApiTokenExistence(t, checkApiTokensExistenceInput{name: apiTokenName, organization: true, shouldExist: false}),
+		),
+		Steps: []resource.TestStep{
+			{
+				Config: astronomerprovider.ProviderConfig(t, true) + apiToken(apiTokenInput{
+					Name:        apiTokenName,
+					Description: utils.TestResourceDescription,
+					Type:        "ORGANIZATION",
```

Comment:
nit we should use `iam.ORGANIZATION` for constants like this and the roles as well
```suggestion
					Type:        "ORGANIZATION",
```